File: go/20. Valid Parentheses.go
package main

func isValid(s string) bool {
	var stack []uint8
	length := len(s)
	stack = append(stack, s[0])
	for i := 1; i < length; i++ {
		if s[i] == '(' || s[i] == '[' || s[i] == '{' {
			stack = append(stack, s[i])
		} else {
			if len(stack) == 0 {
				return false
			}
			if s[i] == ')' && stack[len(stack)-1] == '(' ||
				s[i] == ']' && stack[len(stack)-1] == '[' ||
				s[i] == '}' && stack[len(stack)-1] == '{' {
				stack = stack[:len(stack)-1]
				continue
			}
			return false
		}
	}
	return len(stack) == 0
}

func main() {
	isValid("stewtw")

}

File: js/20. Valid Parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function (s) {
    let stack = [];
    stack.push(s[0]);
    let size = s.length;
    for (let i = 1; i < size; i++) {
        if (s[i] === '(' || s[i] === '[' || s[i] === '{') {
            stack.push(s[i]);
        } else {
            if (stack.length === 0) {
                return false;
            }
            if (s[i] === ')' && stack[stack.length - 1] === '(' ||
                s[i] === ']' && stack[stack.length - 1] === '[' ||
                s[i] === '}' && stack[stack.length - 1] === '{') {
                stack.pop();
                continue;
            }
            return false;
        }
    }
    return stack.length === 0
};